src/server/game/World/Area.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Area.h"
+#include "GameObject.h"
+#include "ObjectMgr.h"
+#include "PoolMgr.h"
+#include "ZoneScript.h"
+
+AreaMgr* AreaMgr::instance()
+{
+    static AreaMgr instance;
+    return &instance;
+}
+
+Area* AreaMgr::GetArea(uint32 areaId)
+{
+    if (!areaId)
+        return nullptr;
+
+    auto itr = m_areas.find(areaId);
+    if (itr != m_areas.end())
+        return itr->second;
+
+    if (AreaTableEntry const* areaEntry = sAreaTableStore.LookupEntry(areaId))
+    {
+        Area* area = new Area(areaEntry);
```

Comment:
The area never gets deleted, maybe use `std::unique_ptr<Area>` as container type

Reply:
I already gave my feedback on IRC about this, its not going to be merged with current design